src/app/[locale]/rest-client/page.tsx

```diff
@@ -27,115 +27,97 @@ export default function RestClientPage() {
     setError(null);
     setResponse(null);
 
+    if (values.contentType === ContentType.JSON && values.data?.trim()) {
+      const isJsonValid = validateJson(values.data, values.contentType);
+
+      if (!isJsonValid) {
+        setError(ERROR_MESSAGES.INVALID_JSON);
+        return;
+      }
+    }
+
+    const headersArray = Array.isArray(values.headers) ? values.headers : [];
+
+    const invalidHeaders = headersArray.filter(
+      (header: Header) =>
+        (header.key.trim() && !header.value.trim()) || (!header.key.trim() && header.value.trim())
+    );
```

Comment:
It would be good to use destructuring here.
Moving trim to a separate variable also looks okay, so as not to do it several times.

```suggestion
    const invalidHeaders = headersArray.filter(
      ({key, value}: Header) => {
        const keyTrim = key.trim();
        const valueTrim = value.trim();

        (keyTrim && !valueTrim) || (!keyTrim && valueTrim)
      }
    );
```

Reply:
Great idea, thank you!

I have applied that change.